lib/mongo/protocol/message.rb

```diff
@@ -226,7 +226,7 @@ def serialize_fields(buffer, max_bson_size = nil)
       # @param buffer [String] Buffer to receive the header
       # @return [String] Serialized header
       def serialize_header(buffer)
-        set_request_id unless @request_id
+        set_request_id unless defined?(@request_id)
```

Comment:
@estolfo, the change you've suggested will not fix a warning "warning: instance variable @request_id not initialized".

Better fix will be to set @request_id in initialize method.